src_erl/NerlnetApp/src/Router/routerGenserver.erl

```diff
@@ -91,18 +92,20 @@ handle_cast({unicast,{Dest,Body}}, State = #router_genserver_state{msgCounter =
   stats:increment_messages_sent(RouterStatsEts),
   {noreply, State#router_genserver_state{msgCounter = MsgCounter+1,etsRef=Routing_table }};
 
-handle_cast({broadcast,{DestList,Body}}, State = #router_genserver_state{etsRef=Routing_table }) ->
+handle_cast({broadcast,{DestList,Body}}, State = #router_genserver_state{etsRef=Routing_table , myName = _MyName}) ->
+  io:format("@ROUTER BOROADCAST  , Body: ~p~n" , [Body]),
   RouterStatsEts = get(router_stats_ets),
+  %%Destinations = [Dest || Dest <- DestList, Dest =/= get(myName)],
   stats:increment_messages_received(RouterStatsEts),
   MapFunc=fun(Dest,Acc)->
     %make a map when keys are addreses to send a message to, and values are lists of destination of the message that go throu key addres
-    [{Dest,{Name,Host,Port}}]=ets:lookup(Routing_table,Dest),
+    [{Dest,{Name,Host,Port}}] = ets:lookup(Routing_table,Dest),
     case maps:is_key({Name,Host,Port},Acc) of
       true->
-        %addres alread in, append Dest to exsisting value 
+        %address already in, append Dest to existing value 
         NewVal=maps:get({Name,Host,Port},Acc)++[Dest];
       false->
-        %addres not in yet, create new value for it
+        %address not in yet, create new value for it
```

Comment:
@GuyPerets106 
Guy please add documentation here for both cases that there is a single returned value from routing table: 
1. Single entity's destination remained from broadcast list for the next hop.
2. entity belongs to current router.